src/core/constants.js

```diff
@@ -48,6 +50,15 @@ export const ADDON_TYPE_THEME = 'persona';
 // TODO: Remove ADDON_TYPE_COMPLETE_THEME once we don't support complete
 // themes.
 export const ADDON_TYPE_COMPLETE_THEME = 'theme';
+
+export type AddonTypeType =
```

Comment:
It seems a shame we have to repeat all of these (both in `AddonTypeType` and in `validAddonTypes`). I think there might be a way to create the type programmatically from the `validAddonTypes` array, rather than having them all repeated like this.

Reply:
Not really yet, because `$Values<>` does not work for JS array. It is mostly due to how Facebook deals with that, i.e. they use objects to describe the actions:

```js
const Actions = {
  SOME_ACTION: 'SOME_ACTION',
  ANOTHER_ONE: 'ANOTHER_ONE',
};
```

With hash maps like the one above, one can use `$Keys<>` or `$Values<>`. Maybe we should be doing this as well..

Reply:
Yeah, I think we're doing that here: https://github.com/mozilla/addons-frontend/blob/72fb82c57cb50815caf16719d30a0d585538cb74/src/ui/components/UserRating/index.js#L19. Is it worth making that change?

Reply:
That would mean changing the code, I'd be in favor of not doing this since `validAddonTypes` is used elsewhere.

Reply:
Fair enough. 👍